packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,234 @@
+// @ts-ignore
+import * as jwt from 'jsonwebtoken';
+import { utils, providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+import { CredentialResolver } from '.';
+import {
+  IVerifiableCredential,
+  VerificationResult,
+  OffChainClaim,
+} from './models';
+
+export class IssuerVerification {
+  private _provider: providers.Provider;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _roleDefResolver: RoleDefinitionResolverV2;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param roleDefResolverAddr
+   * @param registrySetting
+   * @param credentialResolver
+   */
+  constructor({
+    provider,
+    roleDefResolverAddr,
+    registrySetting,
+    credentialResolver,
+  }: {
+    provider: providers.Provider;
+    roleDefResolverAddr: string;
+    registrySetting: RegistrySettings;
```

Comment:
Similar to how we accepting a `CredentialResolver`, I think that `IssuerVerification` should accept a `RoleDefinitionResolver` so that it is independent of role definition resolution implementation.

Reply:
Hmm, sounds good. Or maybe we can do `CachedObjectStrategy` which can accept both `CachedRoleDefinitionResolutionStrategy` and `CachedDidDocumentResolutionStrategy` and resolves the passed objects. Does this sound right?

Reply:
Personally I don't think having `CachedObjectStrategy` is a good idea because it needlessly couples  `CachedRoleDefinitionResolutionStrategy` and `CachedDidDocumentResolutionStrategy`. To me, `CachedObjectStrategy` implies that it is an implementation of `AbstractObjectStrategy` (to use the strategy pattern) and I think `AbstractObjectStrategy` is overly abstract. In other words, an "Object Strategy" is too broad in my view.

Reply:
Hmm, makes sense. I think we now have a better understanding of what we want.